java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/Utils.java

```diff
@@ -426,16 +436,41 @@ public static String html2plain(String s) {
                 } else if (tagStart == -1 && alpha) {
                     tagStart = i;
                 }
+                continue;
             } else {
                 if (ch == '<') { // NOI18N
                     tagStart = -1;
                     inTag = true;
                     continue;
                 }
-                sb.append(ch);
             }
+            if (collapseWhitespaces) {
+                if (Character.isWhitespace(ch)) {
+                    if (whitespace) {
+                        continue;
+                    }
+                    ch = ' '; // NOI18N
+                    whitespace = true;
+                } else {
+                    if (insertWSIfMissing) {
```

Comment:
Is this ever `true`?

Reply:
result of merging two implementations together ;)